.travis.yml

```diff
@@ -15,30 +15,34 @@ jdk:
 # a special Java 8 configuration to build a .jar and upload it somewhere
 matrix:
   include:
-    name: "Build and upload .jar (uploads for master builds and tags only, Java 8)"
-    jdk: openjdk8
-    script:
-      - cd cloud.foundry.cli
-      - ./gradlew jar
-    after_success:
-      - wget -c https://github.com/probonopd/uploadtool/raw/master/upload.sh
-      # upload to alternative location unless on master branch (or building a tag)
-      - |2
-        if [ "$TRAVIS_EVENT_TYPE" == "pull_request" ] || [ "$TRAVIS_BRANCH" != "master" ] && [ "$TRAVIS_TAG" == "" ]; then
-          echo "Uploads to transfer.sh disabled for performance reasons"
-        else
-          bash upload.sh build/libs/cf-control.jar
-        fi
+    - name: "Build and upload .jar (uploads for master builds and tags only, Java 8)"
+      jdk: openjdk8
+      script:
+        - cd cloud.foundry.cli
+        - ./gradlew jar
+      after_success:
+        - wget -c https://github.com/probonopd/uploadtool/raw/master/upload.sh
+        # upload to alternative location unless on master branch (or building a tag)
+        - |2
+          if [ "$TRAVIS_EVENT_TYPE" == "pull_request" ] || [ "$TRAVIS_BRANCH" != "master" ] && [ "$TRAVIS_TAG" == "" ]; then
+            echo "Uploads to transfer.sh disabled for performance reasons"
+          else
+            bash upload.sh build/libs/cf-control.jar
+          fi
+
+    - name: "Run all tests (including system tests) and report coverage (Java 8)"
+      # Really, coveralls plugin? You don't work with Java 11?
+      # https://github.com/kt3k/coveralls-gradle-plugin/issues/85
+      jdk: openjdk8
+      script:
+        - cd cloud.foundry.cli
+        - ./gradlew jacocoTestReport coveralls
```

Comment:
Where is the normal gradle build? Don't we need it before the jacocoTestReport.

Reply:
Not at all. The `jacocoTestReport` command internally depends on the test command, and there's no need to e.g., build a JAR or whatever `build` does. I moved the coverage calculation up to make sure that if it fails, the test run fails, too.

Reply:
Alright thanks :) then it is fine like that.